plugin/src/main/java/org/opensearch/ml/rest/RestMLProfileAction.java

```diff
@@ -154,7 +154,15 @@ public void onResponse(SearchResponse searchResponse) {
 
                     @Override
                     public void onFailure(Exception e) {
-                        onFailed(channel, "Searching model wasn't successful", e);
+                        try {
+                            builder.startObject();
+                            builder.endObject();
+                            channel.sendResponse(new BytesRestResponse(RestStatus.OK, builder));
```

Comment:
Is this same with the old response?

Reply:
Yes. 

```
{}
```

Reply:
I didn't get why we return empty content here if failing to search hidden models. In an example case, super-admin call profile api, we could continue to exeute MLProfileAction to return profiles even if failed to search hidden models, as user is super-admin.

Reply:
This is to keep the experience same. Currently if there's any model in the index/cache, we return an empty object when we call profile api. I'm just keeping the experience same.

Reply:
If failed to search, is it possible the profile API will have other valid response which should be returned to user ? If yes, we should not just return empty `{}`

Reply:
I tested in the newly created cluster (2.13.0) where any model isn't created. If no models in the cluster, we return {}

Reply:
I mean what if for other reason the search failed, not just for newly created cluster which has no model index

Reply:
Even for other reason if the search fails, we need to make sure we aren't exposing any information of hidden model. If the search fails then we don't have any visibility of which one is the hidden model.

Reply:
But super-admin should have permission to get everything even if hidden models, right? We shouldn't block super-admin to get information.

Reply:
Yeah that's true. But super admin won't know which one is the hidden model. Because depending on that search we mark hidden model. If the search isn't successful, then that marking won't happen.